api/graph/model/models_interface.go

```diff
@@ -16,28 +18,53 @@ func RecipeFromDBType(result db.Recipe) *Recipe {
 	return &recipe
 }
 
-func UserFromDBType(result db.User) *User {
-	// Map to model.User type
-	user := User{
-		ID:          int(result.ID),
-		Email:       result.Email,
-		JoinDate:    result.JoinDate.Time,
-		DisplayName: result.DisplayName,
-		UpdatedAt:   result.UpdatedAt.Time,
+func UserFromDBType[T any](result T) *User {
+	switch v := any(result).(type) {
+	case db.User:
+		return &User{
+			ID:          int(v.ID),
+			Email:       v.Email,
+			JoinDate:    v.JoinDate.Time,
+			DisplayName: v.DisplayName,
+			UpdatedAt:   v.UpdatedAt.Time,
+		}
+	case db.GetRecipeWithAuthorByIdRow:
+		return &User{
+			ID:          int(v.UserID),
+			Email:       v.UserEmail,
+			JoinDate:    v.UserJoinDate.Time,
+			DisplayName: v.UserDisplayName,
+			UpdatedAt:   v.UserUpdatedAt.Time,
+		}
```

Comment:
This shouldn't be necessary as the two types are basically the same. We might need to revisit the actual query and make sure we are selecting all fields and that they are in the right order.